cdap-api/src/main/java/io/cdap/cdap/api/app/AppStateStore.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.app;
+
+import java.io.IOException;
+
+/**
+ * Saves and reads state for the given app.
+ */
+public interface AppStateStore {
+
+  /**
+   * Returns the saved state for given app and key.
+   * @param key Key for the state
+   * @return value as byte[]
+   * @throws IOException if the app is not available, key is not available or otherwise unable to fetch state
+   */
+  byte[] getState(String key) throws IOException;
+
+  /**
+   * Saves the state for an app with a key. This state is removed when the app is deleted.
+   * @param key Key for the state
+   * @param value value as byte[]
```

Comment:
is this allowed to be null? If so, should document it explicitly and mention that getState() will return null. Some people may assume that getState() returning null means the key doesn't exist. If not, should document what happens.

Reply:
Done.